executor/aggfuncs/func_max_min.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggfuncs
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+type partialResult4MaxMinInt int64
```

Comment:
How aboult use `type partialResult4MaxMinInt = int64`, 
then we can use `*(*partialResult4MaxMinInt)(pr)`  instead of `int64(*(*partialResult4MaxMinInt)(pr))`

below partialResult4MaxMinDecimal, partialResult4MaxMinFloat32, partialResult4MaxMinFloat64 is same too.